src/NUnitEngine/nunit.engine.core/RuntimeFramework.cs

```diff
@@ -135,6 +138,18 @@ private Version GetClrVersionForFramework(Version frameworkVersion)
                             return new Version(4, 0, 30319);
                     }
                     break;
+                case RuntimeType.NetCore:
+                    switch (frameworkVersion.Major)
+                    {
+                        case 1:
+                        case 2:
+                            return new Version(4, 0, 30319);
```

Comment:
Is this correct? The Azure error looks like it's trying to run the .NET 3.5 tests on .NET 4.X and not finding the framework. This line looks suspicious?

```
1) Error : 
System.ArgumentException : The net-4.0 framework is not available
Parameter name: framework
--ArgumentException
The net-4.0 framework is not available
```

Reply:
Actually, not sure that would make any sense. Still not sure this line is correct, though. 😅

Reply:
According to https://docs.microsoft.com/en-us/dotnet/api/system.environment.version?view=net-5.0, "for ... .NET Core versions before 3.0, [Environment.Version] has the form 4.0.30319.42000."

The current `RuntimeFramework` class assumes that each framework version has a corresponding CLR version. With .NET Core, that no longer makes sense and I think `RuntimeFramework` needs to be rewritten or removed. I didn't do that here, because this PR is already too big. Rather, I tried to avoid using the CLR version to make any decisions.

However, that net-4.0 framework is not available looks suspicious to me too. I'll check into it further.